app/controllers/authorities_controller.rb

```diff
@@ -4,11 +4,31 @@
 class AuthoritiesController < ApplicationController
   def query
     s = params.fetch('q', '')
-    hits = if ['location', 'based_near'].include?(params[:term])
+    term = params[:term]
+    hits = if ['location', 'based_near'].include?(term)
       GeoNamesResource.find_location(s)
+    elsif ['lcsh', 'subject_geographic'].include?(term)
+      get_results(term, s)
     else
       LocalAuthority.entries_by_term(params[:model], params[:term], s) rescue []
     end
     render json: hits
   end
+  
+  def get_results(subject, search)
+    # the /suggest path limits results to 10
+    lcsh_base_uri = "http://id.loc.gov/authorities/subjects/suggest/?q="
+    # can increase limit to max 20 with '&rows=20'
+    fast_geo_uri = "http://fast.oclc.org/searchfast/fastsuggest?fl=suggest51&wt=json&query="
+    if subject == 'lcsh'
+      result = JSON.parse(HTTParty.get(
+        lcsh_base_uri + "#{search.gsub(/\s/,'*')}*"
+      ))[1]
+    elsif subject == 'subject_geographic'
+      result = HTTParty.get(fast_geo_uri + search)["response"]["docs"]
```

Comment:
Is this guaranteed to be an array?

Reply:
you mean in case of one result? i can test for it.